catalog/dags/common/elasticsearch.py

```diff
@@ -0,0 +1,223 @@
+import logging
+from datetime import timedelta
+from typing import Literal, Union
+
+from airflow.decorators import task, task_group
+from airflow.models.connection import Connection
+from airflow.providers.elasticsearch.hooks.elasticsearch import ElasticsearchPythonHook
+from airflow.sensors.python import PythonSensor
+from airflow.utils.trigger_rule import TriggerRule
+
+from common.constants import REFRESH_POKE_INTERVAL
+
+
+logger = logging.getLogger(__name__)
+
+
+# Index settings that should not be copied over from the base configuration when
+# creating a new index.
+EXCLUDED_INDEX_SETTINGS = ["provided_name", "creation_date", "uuid", "version"]
```

Comment:
Nit: This should be a set for check speed and clarity

```suggestion
EXCLUDED_INDEX_SETTINGS = {"provided_name", "creation_date", "uuid", "version"}
```